x-pack/plugins/observability_solution/observability_ai_assistant_app/server/functions/query/validate_esql_query.ts

```diff
@@ -47,15 +48,12 @@ export async function validateEsqlQuery({
     return 'text' in error ? error.text : error.message;
   });
 
-  // With limit 0 I get only the columns, it is much more performant
-  const performantQuery = `${query} | limit 0`;
```

Comment:
I think we also use this in other places, is it safe to change? We should probably also change the name of the function if we actually fetch the rows here as well.

Reply:
I checked that, we dont break anything but we will run a less performant query on the other place without any reason. I followed this path but let me know if you want a different approach https://github.com/elastic/kibana/pull/184106/commits/386471a480c3dc36025afc2004527579a16e0f81

Reply:
I think that we can actually remove the other request in `execute_query` and simply use the same approach as what you've implemented for visualize_query (I'm not sure why we were doing two queries there). Just got to make sure we still send useful error messages to the LLM.

Reply:
Yes this makes sense, I didnt pay a lot of attention to the execute_query function tbh 😄 Done https://github.com/elastic/kibana/pull/184106/commits/7b0e6c41eb170c22968f39b9727bba1fd8b73dc5